src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-join-auth-form/smb-join-auth-form.component.html

```diff
@@ -0,0 +1,126 @@
+<div
+  cdsCol
+  [columnNumbers]="{ md: 4 }"
+>
+  <form name="form"
+        #formDir="ngForm"
+        [formGroup]="form"
+        novalidate>
+    <div i18n="form title"
+         class="form-header">
+      {{ action | titlecase }} {{ resource | upperFirst }}
+    </div>
+
+    <!-- Auth Id -->
+    <div class="form-item">
+      <cds-text-label
+        for="authId"
+        i18n
+        helperText="Unique identifier"
+        i18n-helperText
+        [invalid]="form.controls.authId.invalid && form.controls.authId.dirty"
+        [invalidText]="jaError"
+      >Active directory access resource name
+        <input
+          cdsText
+          type="text"
+          id="authId"
+          name="authId"
+          formControlName="authId"
+          [invalid]="form.controls.authId.invalid && form.controls.authId.dirty"
```

Comment:
I think `showError` checks for both ? may be worth checking if
Similarly for others below

Reply:
With dynamic forms I've had problems with this due to the `this.get(controlName)` not finding the controlname in the form.

I've tried to implement showError instead of current form.control validator but it only accepts `NgForm` as paramter for the form, my form here is a `CdFormGroup`, did not look much else into it, so leaving it as  it is for now